states/statemgr/statemgr_fake.go

```diff
@@ -94,3 +94,56 @@ func (m *fakeFull) Unlock(id string) error {
 	m.locked = false
 	return nil
 }
+
+// NewUnlockErrorFull returns a state manager that is useful for testing errors
+// (mostly Unlock errors) when used with the clistate.Locker interface. Lock()
+// does not return an error because clistate.Locker Lock()s the state at the
+// start of Unlock(), so Lock() must succeeded for Unlock() to get called.
+func NewUnlockErrorFull(t Transient, initial *states.State) Full {
+	if t == nil {
+		t = NewTransientInMemory(nil)
+	}
+
+	// The "persistent" part of our manager is actually just another in-memory
+	// transient used to fake a secondary storage layer.
+	fakeP := NewTransientInMemory(initial.DeepCopy())
+
+	return &fakeErrorFull{
+		t:     t,
+		fakeP: fakeP,
+	}
+}
+
+type fakeErrorFull struct {
+	t     Transient
+	fakeP Transient
+
+	lockLock sync.Mutex
+	locked   bool
```

Comment:
I don't see any of the struct members other than `t` being used in the methods below. Would it make sense to remove them?

Reply:
Honestly I don't think we need _any_ of the struct members, since all the state functions return errors. I'll remove them all!